Spigot-Server-Patches/0408-Add-BellRingEvent.patch

```diff
@@ -0,0 +1,41 @@
+From dc507bbd88bc391eebff1a4c9ccf8c86bf466bfe Mon Sep 17 00:00:00 2001
+From: Eearslya Sleiarion <eearslya@gmail.com>
+Date: Mon, 24 Jun 2019 21:27:32 -0700
+Subject: [PATCH] Add BellRingEvent
+
+Add a new event, BellRingEvent, to trigger whenever a player rings a
+village bell. Passes along the bell block and the player who rang it.
+
+diff --git a/src/main/java/net/minecraft/server/BlockBell.java b/src/main/java/net/minecraft/server/BlockBell.java
+index e83108e3c..248bbb089 100644
+--- a/src/main/java/net/minecraft/server/BlockBell.java
++++ b/src/main/java/net/minecraft/server/BlockBell.java
+@@ -2,6 +2,12 @@ package net.minecraft.server;
+ 
+ import javax.annotation.Nullable;
+ 
++// Paper start
++import org.bukkit.craftbukkit.event.CraftEventFactory;
++import org.bukkit.entity.Player;
++import com.destroystokyo.paper.event.block.BellRingEvent;
++// Paper end
++
+ public class BlockBell extends BlockTileEntity {
+ 
+     public static final BlockStateDirection a = BlockFacingHorizontal.FACING;
+@@ -46,6 +52,12 @@ public class BlockBell extends BlockTileEntity {
+         boolean flag1 = !flag || this.a(iblockdata, enumdirection, movingobjectpositionblock.getPos().y - (double) blockposition.getY());
+ 
+         if (!world.isClientSide && tileentity instanceof TileEntityBell && flag1) {
++            // Paper start - BellRingEvent
++            org.bukkit.block.Block block = world.getWorld().getBlockAt(blockposition.getX(), blockposition.getY(), blockposition.getZ());
++            Player player = (entityhuman == null) ? null : (Player) entityhuman.getBukkitEntity();
++            BellRingEvent event = new BellRingEvent(block, player);
++            if (CraftEventFactory.callEvent(event).isCancelled()) return true;
```

Comment:
Instead of using `CraftEventFactory`, use `Event#callEvent()` which also handles event cancellation already.

`if (!new BellRingEvent(block, player).callEvent()) return true;`